### DISCUSSION TITLE
installing on Rhino (rolling Ubuntu-devel) - How to install cockpit-pcp ?

### DISCUSSION THREAD

**Discussion by iconoclasthero** [cockpit-project/cockpit#20778] [General]
There's no explicit explanation as to why the -t backports apt flag is needed as it isn't explained and I don't remember ever needing to use such a thing in the 15 or so years I've been using `apt`.  Here's the codename for my rolling release:
```bash
 . /etc/os-release; echo "$VERSION_CODENAME" ; uname -a
devel
Linux machine 6.9.1-060901-generic #202405171038 SMP PREEMPT_DYNAMIC Fri May 17 10:55:28 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
```
so when I put that in the apt command:
```bash
$ sudo apt install -t ${VERSION_CODENAME}-backports cockpit
Error: The value 'devel-backports' is invalid for APT::Default-Release as such a release is not available in the sources
```
but since I had the current release version:
```bash
$ apt show cockpit
Package: cockpit
Version: 321-1
Priority: optional
Section: universe/admin
Origin: Ubuntu
⋮
```
it seemed reasonable to install it.  

After reading the linked Ubuntu backport page and guessing as to what was relevant to understand there, I'm going to assume that as I'm running a rolling release, I'm pulling from the backport source anyway, so that's all moot?

If that's the case then the problem I'm having is finding how to install the cockpit-pcp module.
Apparently it tires to use apt, it displays a bunch of warnings/notices from apt (i.e., **not** errors) about disabled PPAs and then the [INSTALL] button is left greyed out...which is fine, but apt doesn't show any cockpit-pcp package to install so I have no idea how to install it...?

So I don't know why the GUI failed since it doesn't see fit to show me what caused it to fail...is it because there's an apt install &&'ed to apt update?  Whatever it is, how do i install cockpit-pcp from the cli?

**Comment by martinpitt** [comment]
Right, the concept of "backports" only exists for *released* Ubuntu versions. It does not apply to a rolling release, so don't use the `-t` option.

Not sure what you mean by the "[INSTALL]" button - you said you tried this on the command line. So does `sudo apt install cockpit` work?

Indeed there is currently no cockpit-pcp. I disabled that for Debian, as [pcp fell out of testing|https://tracker.debian.org/pkg/pcp], and I didn't put in some Ubuntu specific packaging hack to turn it back on there.

**Comment by iconoclasthero** [comment]
tl;dr: Is there a way to install the cockpit-pcp module on Rhino?

> Not sure what you mean by the "[INSTALL]" button - you said you tried this on the command line. So does sudo apt install cockpit work?

at the bottom of https://server:9090/metrics there's a button that says, "Install cockpit-pcp"

```
$ apt install cockpit
cockpit is already the newest version (321-1).
```

I moved /etc/apt/sources.list.d to /etc/apt/sources.list.dumb and tried it again.  I found out that...

> Indeed there is currently no cockpit-pcp. I disabled that for Debian, as [pcp fell out of testing|https://tracker.debian.org/pkg/pcp], and I didn't put in some Ubuntu specific packaging hack to turn it back on there.

yes, so if you asked about `apt install cockpit-pcp` i'd reply that the rhino wrapper for apt and flatpack and pacscripts and whatnot did not find `cockpit-pcp` probably because it's not available (i only tried after apt didn't find anything)...

**Reply by martinpitt** [reply]
Thanks for clarifying. I'll build cockpit-pcp again for the next upload (for anything except debian-testing)

**Reply by iconoclasthero** [reply]
In theory, if you wanted to keep the resulting package out of ubuntu repo—and although I've not actually submitted any pull requests for actual pacscripts—this could be handled via pacstall and an appropriate pacscript.  

I asked ChatGPT to made one for epub-tools and after some work on the pacstall discord, I have something that is more or less correct for a binary install.  It doesn't compile the source though.

**Comment by iconoclasthero** [comment]
```
#!/bin/bash

pkgname="epub-tools-bin"
_pkgname="epub-tools"
pkgver="3.2"
pkgrel="1"
pkgdesc="A set of tools for working with EPUB files"
url="https://github.com/dino-/epub-tools"
arch=('amd64')
depends=()
license=('ISC')
provides=(epub-tools)
conflicts=(epub-tools)
depends=(glibc gmp zlib)
# The URL for the precompiled binary tarball
source=("https://github.com/dino-/${_pkgname}/releases/download/${pkgver}/${_pkgname}-${pkgver}-linux.tar.gz")

sha256sums=("SKIP")

package() {
    cd "${srcdir}"
    tar xzf "epub-tools-3.2-linux.tar.gz"
    cd "epub-tools-3.2"
    install -Dm755 "usr/bin/epubzip" "${pkgdir}/usr/bin/epubzip"
    install -Dm755 "usr/bin/epubname" "${pkgdir}/usr/bin/epubname"
    install -Dm755 "usr/bin/epubmeta" "${pkgdir}/usr/bin/epubmeta"
    install -Dm644 "usr/share/doc/epub-tools/README.md" "${pkgdir}/usr/share/doc/epub-tools/README.md"
    install -Dm644 "usr/share/doc/epub-tools/changelog.md" "${pkgdir}/usr/share/doc/epub-tools/changelog.md"
    install -Dm644 "usr/share/licenses/epub-tools/LICENSE" "${pkgdir}/usr/share/licenses/epub-tools/LICENSE"
}

# Package metadata
pkgmaintainer="Your Name <your.email@example.com>"
pkgsource="https://github.com/dino-/epub-tools"
pkgversion="${pkgver}"
pkgupstream="https://hackage.haskell.org/package/epub-tools"
```
